src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,596 @@
+        """
+        Constructor for the MapperComplex class.
+
+        Parameters:
+            input_type (string): type of input data. Either "point cloud" or "distance matrix".
+            min_points_per_node (int): threshold on the size of the cover complex nodes (default 0). Any node associated to a subpopulation with less than **min_points_per_node** points will be removed.
+            filter_bnds (list of lists or numpy array of shape (num_filters) x 2): limits of each filter, of the form [[f_1^min, f_1^max], ..., [f_n^min, f_n^max]]. If one of the values is numpy.nan, it can be computed from the dataset with the fit() method.
+            resolutions (list or numpy array of shape num_filters containing integers): resolution of each filter function, ie number of intervals required to cover each filter image. If None, it is estimated from data.
+            gains (list or numpy array of shape num_filters containing doubles in [0,1]): gain of each filter function, ie overlap percentage of the intervals covering each filter image. If None, it is set as 0.33, since in the automatic parameter selection method in http://www.jmlr.org/papers/volume19/17-291/17-291.pdf, any arbitrary value between 1/3 and 1/2 works, so we go with the minimal one (ensuring that the complex is a graph if only given one filter).
+            clustering (class): clustering class (default sklearn.cluster.DBSCAN()). Common clustering classes can be found in the scikit-learn library (such as AgglomerativeClustering for instance). If None, it is set to hierarchical clustering, with scale estimated from data.
+            N (int): subsampling iterations (default 100) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            beta (double): exponent parameter (default 0.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.) for estimating scale and resolutions. Used only if clustering or resolutions = None. See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            verbose (bool): whether to display info while computing.
+        """
+
+        self.filter_bnds, self.resolutions, self.gains, self.clustering = filter_bnds, resolutions, gains, clustering
+
+        if self.resolutions is not None:
+            self.resolutions = np.array(self.resolutions)
+            if len(self.resolutions.shape) == 0:
+                self.resolutions = np.broadcast_to(self.resolutions, 1)
+        if self.gains is not None:
+            self.gains = np.array(self.gains)
+            if len(self.gains.shape) == 0:
+                self.gains = np.broadcast_to(self.gains, 1)
+        if self.filter_bnds is not None:
+            self.filter_bnds = np.array(self.filter_bnds)
```

Comment:
The scikit-learn recommendation (https://scikit-learn.org/stable/developers/develop.html#instantiation) is that `__init__` should be trivial (only store the arguments), and that all this kind of logic should be moved to `fit`. We don't have to follow it, but if we don't, we should not derive from BaseEstimator because it won't work properly.

Reply:
Done!